drivers/lte_link_control/lte_lc.c

```diff
@@ -296,6 +311,99 @@ int lte_lc_psm_req(bool enable)
 	return 0;
 }
 
+int lte_lc_psm_get(int *tau, int *active_time)
+{
+	int err;
+	struct at_param_list at_resp_list = {0};
+	char buf[AT_CEREG_RESPONSE_MAX_LEN] = {0};
+	char timer_str[9] = {0};
+	char unit_str[4] = {0};
+	size_t timer_str_len = sizeof(timer_str) - 1;
+	size_t unit_str_len = sizeof(unit_str) - 1;
+	size_t index;
+	u32_t timer_unit, timer_value;
+
+	/* Enable network registration status with PSM information */
+	err = at_cmd_write(AT_CEREG_5, NULL, 0, NULL);
+	if (err) {
+		LOG_ERR("Could not set CEREG, error: %d", err);
+		return err;
+	}
+
+	/* Read network registration status */
+	err = at_cmd_write(AT_CEREG_READ,	 buf, sizeof(buf), NULL);
+	if (err) {
+		LOG_ERR("Could not get CEREG response, error: %d", err);
+		return err;
+	}
+
+	err = at_params_list_init(&at_resp_list, AT_CEREG_PARAMS_COUNT);
+	if (err) {
+		LOG_ERR("Could not init AT params list, error: %d", err);
+		return err;
+	}
+
+	err = at_parser_max_params_from_str(buf,
+					    NULL,
+					    &at_resp_list,
+					    AT_CEREG_PARAMS_COUNT);
+	if (err) {
+		LOG_ERR("Could not parse AT+CEREG response, error: %d", err);
+		goto parse_psm_clean_exit;
+	}
+
+	/* Parse periodic TAU string */
+	err = at_params_string_get(&at_resp_list,
+				   AT_CEREG_TAU_INDEX,
+				   timer_str,
+				   &timer_str_len);
+	if (err) {
+		LOG_ERR("Could not get TAU, error: %d", err);
+		goto parse_psm_clean_exit;
+	}
+
+	memcpy(unit_str, timer_str, unit_str_len);
+
+	index = strtoul(unit_str, NULL, 2);
+	if (index > (ARRAY_SIZE(t3412_lookup) - 1)) {
+		LOG_ERR("Unable to parse periodic TAU string");
+		return -EINVAL;
```

Comment:
`goto parse_psm_clean_exit;`? Should set `err` as well.

Reply:
Yep, fixed now